src/ui/components/results/resultsheadercomponent.js

```diff
@@ -86,6 +86,21 @@ export default class ResultsHeaderComponent extends Component {
     });
   }
 
+  /**
+   * Returns the currently applied nlp filter nodes, with nlp filter nodes that
+   * are duplicates of other static filter nodes removed.
+   * @returns {Array<FilterNode>}
+   */
+  _pruneDuplicateNlpFilterNodes () {
+    return this.nlpFilterNodes.filter(filterNode => {
+      const isDuplicate = this.appliedFilterNodes.find(node =>
+        filterNode.getFilter().getFilterKey() === node.getFilter().getFilterKey() &&
+        filterNode.getMetadata().displayValue === node.getMetadata().displayValue
+      );
```

Comment:
Is this the only way to check that two nodes are the "same"? It makes me a bit nervous that we're doing a comparison based on something like displayValue. That's not always the best indicator of two objects sharing the same value.

Reply:
you're right, filters should probably have some kind of equals method

Reply:
Tried to add some lodash helpers for this, which increased the bundle size by a few kb.

```

  dist/answers.min.js
  Package size limit has exceeded by 4.42 KB
  Package size: 154.42 KB
  Size limit:   150 KB

  dist/answers-modern.min.js
  Package size limit has exceeded by 2.93 KB
  Package size: 102.93 KB
  Size limit:   100 KB

  dist/answerstemplates.compiled.min.js
  Package size: 58.83 KB
  Size limit:   100 KB
```

Before this change, the bundle size was

```
  dist/answers.min.js
  Package size: 148.96 KB
  Size limit:   150 KB

  dist/answers-modern.min.js
  Package size: 95.3 KB
  Size limit:   100 KB
 
  dist/answerstemplates.compiled.min.js
  Package size: 59.14 KB
  Size limit:   100 KB
 
  With all dependencies, minified and gzipped
```